packages/core/src/components/button/buttons.tsx

```diff
@@ -8,107 +8,52 @@
 // HACKHACK: these components should go in separate files
 // tslint:disable max-classes-per-file
 
-import * as classNames from "classnames";
 import * as React from "react";
 
-import * as Classes from "../../common/classes";
 import { removeNonHTMLProps } from "../../common/props";
-import { Spinner } from "../spinner/spinner";
-import { AbstractButton, IButtonProps } from "./abstractButton";
+
+// namespace import for re-exported IButtonProps ("cannot be named")
```

Comment:
will add code comments. basically `IButtonProps` is used in the exported component type, but never declared in this file. if I import it here, there's a `tsc` error that it's unused (cuz it's never explicitly referenced in this file). and if i don't import it, then there's an error on the Factory line about "cannot be named." catch-22 indeed.

namespace import is the only solution that produces no errors.